library/core/src/panic/assert_info.rs

```diff
@@ -0,0 +1,52 @@
+use crate::fmt::Debug;
+
+/// Information about a failed assertion.
+#[derive(Debug)]
+pub struct AssertInfo<'a> {
+    /// The assertion that failed.
+    pub assertion: Assertion<'a>,
+    /// Optional additional message to include in the failure report.
+    pub message: Option<&'a crate::fmt::Arguments<'a>>,
```

Comment:
It's probably best to use `Option<Arguments>` here instead of `Option<&Arguments>`. The `Arguments` will probably just end up on the stack next to the `AssertInfo`, so might as well be inlined into the object directly.